### PR TITLE
Issue 12883 additional arguments for summary image

### PR DESCRIPTION
**Pull request by feihugis**
This PR aims to add `vmin`, `vmax`, and `clip` parameters for summary image op. More details can be found at the issue-12883.

### CODE REVIEW

File: tensorflow/core/kernels/summary_image_op.cc [hunk 1]
```diff
@@ -32,10 +32,28 @@ class SummaryImageOp : public OpKernel {
...
+    OP_REQUIRES_OK(context, context->GetAttr("clip", &clip_tmp));
+
+    vmin_ = static_cast<float>(vmin_tmp);
+    vmax_ = static_cast<float>(vmax_tmp);
+    clip_ = static_cast<bool>(clip_tmp);
+
+    if (vmin_ != default_val_ && vmax_ != default_val_) {
+      OP_REQUIRES(context, vmin_ <= vmax_,
+                  errors::InvalidArgument("vmin must be <= vmax"));
+    }
```
Comment (frreiss) [inline tensorflow/core/kernels/summary_image_op.cc]: This check will fail if the user specifies `vmax` but not `vmin`.

File: tensorflow/core/ops/logging_ops.cc [hunk 2]
```diff
@@ -80,6 +80,9 @@ REGISTER_OP("ImageSummary")
     .Input("tag: string")
     .Input("tensor: T")
     .Output("summary: string")
+    .Attr("vmin: float = 9999.99")
```
Comment (frreiss) [inline tensorflow/core/ops/logging_ops.cc]: This default value may not be large enough; some images have 16 bits per channel per pixel.

File: tensorflow/python/summary/summary_test.py [hunk 3]
```diff
@@ -100,7 +100,8 @@ def testImageSummaryWithFamily(self):
     with self.cached_session() as s:
       i = array_ops.ones((5, 2, 3, 1))
       with ops.name_scope('outer'):
-        im = summary_lib.image('inner', i, max_outputs=3, family='family')
+        im = summary_lib.image('inner', i, max_outputs=3, vmin=1.0, vmax=1.0,
```
Comment (frreiss) [inline tensorflow/python/summary/summary_test.py:103]: You should also test the`vmin < vmax`, `vmin` only, and `vmax` only cases.

File: tensorflow/python/summary/summary.py [hunk 4]
```diff
@@ -129,6 +129,10 @@ def image(name, tensor, max_outputs=3, collections=None, family=None):
     tensor: A 4-D `uint8` or `float32` `Tensor` of shape `[batch_size, height,
       width, channels]` where `channels` is 1, 3, or 4.
     max_outputs: Max number of batch elements to generate images for.
+    vmin: Maximum clipping value.
```
Comment (frreiss) [inline tensorflow/python/summary/summary.py:136]: Recommend a more verbose comment, along the lines of how matplotlib describes vmin and vmax in https://matplotlib.org/devdocs/api/_as_gen/matplotlib.pyplot.imshow.html
Also describe what happens if the user specifies only one of these two parameters.

**Review by frreiss — COMMENTED** [review]
Looks good overall, some minor suggestions added.

File: tensorflow/python/summary/summary_test.py (hunk 3 above)
Comment (feihugis) [reply tensorflow/python/summary/summary_test.py:103]: Have added more test cases.

File: tensorflow/python/summary/summary.py (hunk 4 above)
Comment (feihugis) [reply tensorflow/python/summary/summary.py:136]: yeah, add more description at the comments.

**Comment by feihugis** [comment]
@frreiss Thanks for your comments! The code has been updated.

File: tensorflow/core/kernels/summary_image_op_test.cc [hunk 5]
```diff
@@ -57,9 +57,25 @@ class SummaryImageOpTest : public OpsTestBase {
...
+                     .Attr("vmax", vmax)
+                     .Attr("clip", clip)
+                     .Finalize(node_def()));
+    TF_ASSERT_OK(InitOp());
+  }
+
   void CheckAndRemoveEncodedImages(Summary* summary) {
     for (int i = 0; i < summary->value_size(); ++i) {
       Summary::Value* value = summary->mutable_value(i);
+      TF_CHECK_OK(WriteStringToFile(
```
Comment (frreiss) [inline tensorflow/core/kernels/summary_image_op_test.cc:75]: This looks like debug code.

**Review by frreiss — APPROVED** [review]
Looks ready to submit the PR, except one apparent bit of debugging code left in

File: tensorflow/core/kernels/summary_image_op_test.cc (hunk 5 above)
Comment (feihugis) [reply tensorflow/core/kernels/summary_image_op_test.cc:75]: Thanks. Have removed it.

**Pull request closed without merging**